### DISCUSSION TITLE
Filter with Reverse / Indirect Relationship

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#3037] [Q&A]
Lets say there are two tables, Invoices & Invoice Items
**Invoices**
- Date
- Customer(Relationship field of customer)
- invoiceItems (Multi Relational field of InvoiceItem table)
- Amount
**InvoiceItems**
- Product(Relationship field of Product)
- Rate
- Quantity
- Amount

In above, I want to get latest rate of product invoiced to a particular customer, so how to filter it?

If I want to get only for product then it can be done by
await pb.collection("InvoiceItems").getList(perPage: 1,sort: "-created",filter: 'Product="ae40239d2bc4477"')

But how to get rate for particular Customer which is in invoices table?

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the requirements, but indirect filter and sort are currently not supported with the web APIs (_it is in the [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=17189183) but currently the SDKs refactoring are with higher priority_).

You can either try creating a helper view collection, send separate requests OR change the way how you query your data (eg. start from `invoices`).

**Comment by lbensaad** [comment]
I have the same tables as you and I might need that filter. I am still working on other things but I might need the `lastProductCustomerPrice` information. when I need it, I will not make it a query because the `invoiceItems` is the biggest table in terms of number of records and  it will be resource-consuming to sort and filter it.
I will rather implement it like this: 
Add another table called `lastPrice (id, custmer (a rlation),  product (a relation), price)`. Whenever a customer buys a product I update the `lastPrice` table. When need the `lastProductCustomerPrice` I will just query the `lastPrice` table
